### PR TITLE
sync the xdp example with generated code

### PR DESCRIPTION
**Pull request by dmitris**
PR synchronizes the "hello-xdp" example with the code currently generated by `cargo generate`.  It used to be that the code changes describe in the current version of https://aya-rs.github.io/book/start/hello-xdp.html were required to get the program running but currently the generated code is sufficient to run the "accept all" XDP program.  Therefore I deleted code under https://github.com/aya-rs/book/tree/main/examples/myapp-01 that contained those changes.

### CODE REVIEW

File: src/start/hello-xdp.md [hunk 1]
```diff
@@ -77,44 +78,60 @@ The generated application has the following content:
...
 
-Finally, we can attach it to an interface with `probe.attach(&iface, XdpFlags::default())?;`
+Finally, we can attach it to an interface with `program.attach(&opt.iface, XdpFlags::default())?;`
 
 Let's try it out!
 
 ```console
 $ cargo build
-$ sudo ./target/debug/myapp --path ./target/bpfel-unknown-none/debug/myapp wlp2s0
+$ sudo ./target/debug/myapp -h
```
Comment (dave-tucker) [inline src/start/hello-xdp.md:119]: You should replace this with `cargo xtask run`

File: src/start/hello-xdp.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ We must first write the eBPF component of our program.
 The logic for this program is located in `myapp-ebpf/src/main.rs` and currently looks like this:
 
 ```rust,ignore
-{{#rustdoc_include ../../examples/myapp-00/myapp-ebpf/src/main.rs}}
+{{#rustdoc_include ../../examples/myapp-00/myapp-ebpf/src/main.rs:all}}
```
Comment (dave-tucker) [inline src/start/hello-xdp.md:17]: this isn't required

**Review by dave-tucker — CHANGES_REQUESTED** [review]
To make the diff easier to review could you please:

1. `mv myapp-00/* myapp-01 && rm -r myapp-00`
2. sed -i `s/myapp-00/myapp-01/g`

Basically `myapp-00` should ne deleted and the first example is now `myapp-01`

The wording changes LGTM, but please address the comments I've left and squash your commits.

**Comment by dmitris** [comment]
@dave-tucker thanks for the review - address the comments on `myapp-00`/`myapp-01`, but didn't get what you suggest about the anchors.  If I don't use `:all` in src/start/hello-xdp.md:17 `{{#rustdoc_include ../../examples/myapp-01/myapp-ebpf/src/main.rs:all}}`, it starts to show all the `// ANCHOR` comments.  Am I doing something wrong?  Do you mean I shouldn't have the `//ANCHOR: all` around the whole file?

File: src/start/hello-xdp.md (hunk 1 above)
Comment (dmitris) [reply src/start/hello-xdp.md]: I would think that `sudo ./target/debug/myapp -h` is more "explicit" and clear ("we envoke the help options for the executable") than `cargo xtask run -- -h`, no? 😄  In any case, changed to `cargo xtask run`.

**Comment by dave-tucker** [comment]
@dmitris you're right. my mistake.
`{{#rustdoc_include ../../examples/myapp-01/myapp-ebpf/src/main.rs:all}}` is fine

File: src/start/hello-xdp.md [hunk 3]
```diff
@@ -67,54 +68,71 @@ Simple!
...
+
+FLAGS:
+    -h, --help       Prints help information
+    -V, --version    Prints version information
+
+OPTIONS:
+    -i, --iface <iface>     [default: eth0]
+
+# add -i or --iface parameter if need to attach to an interface other than eth0
+# - for example, cargo xtask run -- -h -- -i wlp2s0
```
Comment (dave-tucker) [inline src/start/hello-xdp.md:134]: This should be `cargo xtask run -- -i wlp2s0`

Comment (dmitris) [reply src/start/hello-xdp.md]: fixed in 425f4b0

File: src/start/logging-packets.md [hunk 4]
```diff
@@ -41,7 +41,7 @@ When the eBPF program is loaded by Aya, the map will be created for us.
 
 ### Userspace: Map Creation
 
-After our call to `probe.attach()` we'll add the following code.
+After our call to `probe.attach()` we'll add the following code:
```
Comment (dave-tucker) [inline src/start/logging-packets.md:44]:
```suggestion
After our call to `program.attach()` we'll add the following code:
```

**Comment by dave-tucker** [comment]
Thanks @dmitris, LGTM

**Pull request merged**